<a href="https://colab.research.google.com/github/Sedowlas/EDA/blob/main/Olympic_game_SQL.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Колонка медали - это категориальная колонка со значениями и ссылками:  
0 - Нет медали  
1 - Бронзовая медаль  
2 - Серебряная медаль  
3 - Золотая медаль  



https://drive.google.com/file/d/1iEWQ5Ybxq5I0D0sYGBhIXBeihGHfQLMn/view?usp=sharing

#Предобработка

In [ ]:
! gdown --id 1iEWQ5Ybxq5I0D0sYGBhIXBeihGHfQLMn

/usr/local/lib/python3.7/dist-packages/gdown/cli.py:131: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  category=FutureWarning,
Downloading...
From: https://drive.google.com/uc?id=1iEWQ5Ybxq5I0D0sYGBhIXBeihGHfQLMn
To: /content/All Year Olympic Dataset (with 2020 Tokyo Olympics).csv
100% 22.1M/22.1M [00:00<00:00, 73.3MB/s]


In [ ]:
!gdown --id 1BSHIKQ7rFw5BpTq5nw1UZfjPK_7Mpnbi
!mv _sqlite3.cpython-37m-x86_64-linux-gnu.so /usr/lib/python3.7/lib-dynload/
import os
os.kill(os.getpid(), 9)

/usr/local/lib/python3.7/dist-packages/gdown/cli.py:131: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  category=FutureWarning,
Downloading...
From: https://drive.google.com/uc?id=1BSHIKQ7rFw5BpTq5nw1UZfjPK_7Mpnbi
To: /content/_sqlite3.cpython-37m-x86_64-linux-gnu.so
100% 6.50M/6.50M [00:00<00:00, 130MB/s]


In [ ]:
import pandas as pd
import numpy as np

In [ ]:
import sqlite3

In [ ]:
con = sqlite3.connect('db')

In [ ]:
df = pd.read_csv('/content/All Year Olympic Dataset (with 2020 Tokyo Olympics).csv')

In [ ]:
df.head()

Unnamed: 0                      Name Sex   Age            Team  \
0           0                 A Dijiang   M  24.0           China   
1           1                  A Lamusi   M  23.0           China   
2           2       Gunnar Nielsen Aaby   M  24.0         Denmark   
3           3      Edgar Lindenau Aabye   M  34.0  Denmark/Sweden   
4           4  Christine Jacoba Aaftink   F  21.0     Netherlands   

         Games  Year  Season          Sport  Medal  
0  1992 Summer  1992  Summer     Basketball      0  
1  2012 Summer  2012  Summer           Judo      0  
2  1920 Summer  1920  Summer       Football      0  
3  1900 Summer  1900  Summer     Tug-Of-War      3  
4  1988 Winter  1988  Winter  Speed Skating      0

In [ ]:
len(df)

275643

In [ ]:
df.drop('Unnamed: 0',axis=1,inplace=True)

In [ ]:
df.columns = df.columns.str.lower()

In [ ]:
df.isna().mean()

name      0.000000
sex       0.000000
age       0.034371
team      0.000000
games     0.000000
year      0.000000
season    0.000000
sport     0.000000
medal     0.000000
dtype: float64

In [ ]:
df.to_sql('olympic',con,index=False,if_exists='replace')

In [ ]:
def select(sql):
  return pd.read_sql(sql,con)

#Анализ

In [ ]:
sql = '''select * from olympic t limit 5'''

In [ ]:
select(sql)

name sex   age            team        games  year  \
0                 A Dijiang   M  24.0           China  1992 Summer  1992   
1                  A Lamusi   M  23.0           China  2012 Summer  2012   
2       Gunnar Nielsen Aaby   M  24.0         Denmark  1920 Summer  1920   
3      Edgar Lindenau Aabye   M  34.0  Denmark/Sweden  1900 Summer  1900   
4  Christine Jacoba Aaftink   F  21.0     Netherlands  1988 Winter  1988   

   season          sport  medal  
0  Summer     Basketball      0  
1  Summer           Judo      0  
2  Summer       Football      0  
3  Summer     Tug-Of-War      3  
4  Winter  Speed Skating      0

In [ ]:
pd.set_option('display.max_rows', None)

In [ ]:
sql = '''select t.team,
count(1) as cnt,
avg(t.medal) as mean_avg

from prison t
group by t.team
'''

In [ ]:
select(sql)

team    cnt  mean_avg
0                                         30. Februar      2  0.000000
1                               A North American Team      4  1.000000
2                                           Acipactli      3  0.000000
3                                             Acturus      2  0.000000
4                                         Afghanistan    128  0.015625
5                                            Akatonbo      3  0.000000
6                                            Alain IV      3  0.000000
7                                             Albania     70  0.000000
8                                              Alcaid      3  0.000000
9                                            Alcyon-6      1  0.000000
10                                           Alcyon-7      2  0.000000
11                                          Aldebaran      2  0.000000
12                                       Aldebaran II      2  0.000000
13                                             Aletta      3  0.000000
14                                            Algeria    607  0.087315
15                                        Ali-Baba II      7  0.714286
16                                        Ali-Baba IV      2  0.000000
17                                        Ali-Baba IX      2  0.000000
18                                        Ali-Baba VI      2  0.000000
19                                            Allegro      3  0.000000
20                                              Almaz      2  0.000000
21                                           Aloha II      5  0.000000
22                       Amateur Athletic Association      5  3.000000
23                                     American Samoa     37  0.000000
24                                          Amolgavar      3  0.000000
25                                   Amstel Amsterdam      4  1.000000
26                                           Amulet-3      4  0.000000
27                                           Amulet-7      2  0.000000
28                                             Ancora      4  3.000000
29                                          Andorinha      2  0.000000
30                                            Andorra    169  0.000000
31                                          Andromeda      3  0.000000
32                                           Angelita     18  2.000000
33                                          Angerburg      1  0.000000
34                                             Angola    269  0.000000
35                                             Anitra      6  0.000000
36                                            Antares      2  0.000000
37                                Antigua and Barbuda    133  0.000000
38                                        Antwerpia V     10  0.500000
39                                             Apache      5  0.000000
40                                          Aphrodite      3  1.000000
41                                           Arcturus      2  0.000000
42                                            Ardilla      3  0.000000
43                                            Aretusa      3  0.000000
44                                          Argentina   3258  0.173112
45                                        Argentina-1     17  0.000000
46                                        Argentina-2     17  0.000000
47                                            Argeste      3  0.000000
48                                            Argo II      3  0.000000
49                               Argonaut Rowing Club      6  1.000000
50                                              Argus      4  0.000000
51                                              Ariel      2  0.000000
52                                         Ariette-10      2  0.000000
53                                          Ariette-8      1  0.000000
54                                            Armenia    222  0.112613
55                                              Aruba     43  0.000000
56

Узнали топ-3 стран с большим количестовом медалей:   
1.United States--17847  
2.France--12134  
3.Italy--10374

In [ ]:
sql = '''select t.games,
count(1) as cnt,
avg(t.age) as mean_avg

from prison t
group by t.games
'''

In [ ]:
select(sql)

games    cnt   mean_avg
0   1896 Summer    380  23.580645
1   1900 Summer   1936  29.034031
2   1904 Summer   1301  26.698150
3   1906 Summer   1733  27.125253
4   1908 Summer   3101  26.970228
5   1912 Summer   4040  27.538620
6   1920 Summer   4292  29.290978
7   1924 Summer   5233  28.449373
8   1924 Winter    460  27.590571
9   1928 Summer   4992  29.414907
10  1928 Winter    582  26.581301
11  1932 Summer   2969  33.451540
12  1932 Winter    352  25.547112
13  1936 Summer   6506  27.878331
14  1936 Winter    895  25.048643
15  1948 Summer   6405  29.304796
16  1948 Winter   1075  26.239029
17  1952 Summer   8270  26.258851
18  1952 Winter   1088  25.446691
19  1956 Summer   5127  26.033673
20  1956 Winter   1307  25.549922
21  1960 Summer   8119  25.298760
22  1960 Winter   1116  24.241877
23  1964 Summer   7702  25.065283
24  1964 Winter   1778  24.419830
25  1968 Summer   8588  24.255154
26  1968 Winter   1891  24.215812
27  1972 Summer  10304  24.341103
28  1972 Winter   1655  24.107748
29  1976 Summer   8641  23.863837
30  1976 Winter   1861  23.739459
31  1980 Summer   7191  23.704354
32  1980 Winter   1746  23.656160
33  1984 Summer   9454  24.002811
34  1984 Winter   2134  23.443241
35  1988 Summer  12037  24.132177
36  1988 Winter   2639  23.840607
37  1992 Summer  12977  24.396165
38  1992 Winter   3436  24.027948
39  1994 Winter   3160  24.422103
40  1996 Summer  13780  24.915045
41  1998 Winter   3605  25.163197
42  2000 Summer  13821  25.422504
43  2002 Winter   4109  25.916281
44  2004 Summer  13443  25.639515
45  2006 Winter   4382  25.959151
46  2008 Summer  13602  25.734118
47  2010 Winter   4402  26.124262
48  2012 Summer  12920  25.961378
49  2014 Winter   4891  25.987324
50  2016 Summer  13688  26.207919
51  2020 Summer   4527  31.463883

Узнали что больше всего спортсменов учавствовало летом 2000 года (13821-спортсмен) и их средний возраст составляет 25 лет 

In [ ]:
sql = '''select t.games,
count(1) as cnt,
avg(t.medal) as mean_avg

from prison t
group by t.games
'''

In [ ]:
select(sql)

games    cnt  mean_avg
0   1896 Summer    380  0.815789
1   1900 Summer   1936  0.637397
2   1904 Summer   1301  0.764796
3   1906 Summer   1733  0.535488
4   1908 Summer   3101  0.548210
5   1912 Summer   4040  0.472277
6   1920 Summer   4292  0.638863
7   1924 Summer   5233  0.318555
8   1924 Winter    460  0.604348
9   1928 Summer   4992  0.293069
10  1928 Winter    582  0.304124
11  1932 Summer   2969  0.444257
12  1932 Winter    352  0.534091
13  1936 Summer   6506  0.284507
14  1936 Winter    895  0.242458
15  1948 Summer   6405  0.267603
16  1948 Winter   1075  0.246512
17  1952 Summer   8270  0.217654
18  1952 Winter   1088  0.248162
19  1956 Summer   5127  0.349132
20  1956 Winter   1307  0.230298
21  1960 Summer   8119  0.224535
22  1960 Winter   1116  0.264337
23  1964 Summer   7702  0.267723
24  1964 Winter   1778  0.210911
25  1968 Summer   8588  0.246274
26  1968 Winter   1891  0.212057
27  1972 Summer  10304  0.234375
28  1972 Winter   1655  0.242900
29  1976 Summer   8641  0.304363
30  1976 Winter   1861  0.226760
31  1980 Summer   7191  0.383257
32  1980 Winter   1746  0.249141
33  1984 Summer   9454  0.311720
34  1984 Winter   2134  0.208060
35  1988 Summer  12037  0.260447
36  1988 Winter   2639  0.198939
37  1992 Summer  12977  0.260384
38  1992 Winter   3436  0.184517
39  1994 Winter   3160  0.208861
40  1996 Summer  13780  0.265820
41  1998 Winter   3605  0.242718
42  2000 Summer  13821  0.288763
43  2002 Winter   4109  0.233390
44  2004 Summer  13443  0.296734
45  2006 Winter   4382  0.240301
46  2008 Summer  13602  0.298265
47  2010 Winter   4402  0.236938
48  2012 Summer  12920  0.296827
49  2014 Winter   4891  0.244940
50  2016 Summer  13688  0.292811
51  2020 Summer   4527  0.726530

Видим, что больше всего медалей спортсмены получили в 2000 году, летом.

In [ ]:
sql = '''select t.sport,
count(1) as cnt,
avg(t.medal) as mean_avg

from prison t
group by t.sport
'''

In [ ]:
select(sql)

sport    cnt  mean_avg
0                 Aeronautics      1  3.000000
1               Alpine Skiing   8829  0.097180
2                    Alpinism     25  3.000000
3                     Archery   2473  0.325920
4            Art Competitions   3578  0.086641
5                   Athletics  39766  0.221018
6                   Badminton   1547  0.265676
7                    Baseball    894  0.751678
8                  Basketball   4536  0.478175
9               Basque Pelota      2  3.000000
10           Beach Volleyball    564  0.255319
11                   Biathlon   4893  0.166973
12                  Bobsleigh   3058  0.259320
13                     Boccia    114  0.649123
14                     Boxing   6047  0.280966
15               Canoe Sprint     91  0.538462
16                   Canoeing   6171  0.377410
17                    Cricket     24  2.500000
18                    Croquet     19  0.947368
19       Cross Country Skiing   9133  0.169167
20                    Curling    463  0.658747
21                    Cycling  10859  0.233539
22               Cycling Road    213  0.906103
23              Cycling Track    115  1.113043
24                     Diving   2842  0.300141
25                 Equestrian     77  0.519481
26              Equestrianism   6344  0.305013
27                    Fencing  10735  0.327340
28             Figure Skating   2298  0.337685
29                   Football   6745  0.461675
30          Football 5-a-side     78  0.769231
31           Freestyle Skiing    937  0.217716
32                   Goalball    118  0.610169
33                       Golf    247  0.368421
34                 Gymnastics  26707  0.171640
35                   Handball   3665  0.577353
36                     Hockey   5417  0.565258
37                 Ice Hockey   5516  0.554931
38               Jeu De Paume     11  0.545455
39                       Judo   3937  0.266193
40                   Lacrosse     60  2.200000
41                       Luge   1479  0.244760
42        Military Ski Patrol     24  1.000000
43          Modern Pentathlon   1677  0.221825
44               Motorboating     17  1.235294
45            Nordic Combined   1344  0.245536
46                       Polo     95  1.431579
47               Powerlifting    178  0.674157
48                   Racquets     12  1.583333
49        Rhythmic Gymnastics    658  0.401216
50                      Roque      4  1.500000
51                     Rowing  10703  0.554237
52                      Rugby    162  2.314815
53               Rugby Sevens    299  0.494983
54                    Sailing   6586  0.386729
55                   Shooting  11602  0.217376
56  Short Track Speed Skating   1534  0.370926
57         Sitting Volleyball    187  0.764706
58                   Skeleton    199  0.301508
59                Ski Jumping   2401  0.172845
60               Snowboarding    936  0.192308
61                   Softball    478  0.751046
62              Speed Skating   5613  0.207198
63                   Swimming  23799  0.285810
64      Synchronized Swimming    909  0.477448
65               Table Tennis   2233  0.260188
66                  Taekwondo    677  0.451994
67                     Tennis   2862  0.229909
68               Trampolining    152  0.394737
69                  Triathlon    609  0.177340
70                 Tug-Of-War    170  1.435294
71                 Volleyball   3404  0.568449
72                 Water Polo   3846  0.547062
73              Weightlifting   3937  0.328423
74      Wheelchair Basketball    262  0.549618
75         Wheelchair Fencing     96  0.916667
76           Wheelchair Rugby     92  0.782609
77          Wheelchair Tennis    104  0.403846
78                  Wrestling   7154  0.354627

Узнали топ-3 вида спорта где учавствовало больше всего спортсменов:  

             кол.участников     процент выйгранных медалей    
      
1.Athletics-------39766--------------------0.221018   
  
2.Gymnastic----26707--------------------0.171640   
  
3.Swimming-----23799--------------------0.285810
